File: app/src/main/java/com/lamnt/foodorder/view/activity/MainActivity.java
package com.lamnt.foodorder.view.activity;

import android.os.Bundle;
import android.view.MenuItem;
import android.widget.ImageView;
import android.widget.RelativeLayout;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.Fragment;
import androidx.swiperefreshlayout.widget.SwipeRefreshLayout;

import com.google.android.material.badge.BadgeDrawable;
import com.google.android.material.bottomnavigation.BottomNavigationView;
import com.lamnt.foodorder.R;
import com.lamnt.foodorder.utils.FragmentUtil;
import com.lamnt.foodorder.view.fragment.foody.HomeFragment;
import com.lamnt.foodorder.view.fragment.foody.OrderFragment;
import com.lamnt.foodorder.view.fragment.payment.PaymentFragment;
import com.lamnt.foodorder.view.fragment.foody.RestaurantFragment;

import butterknife.ButterKnife;
import butterknife.OnClick;

public class MainActivity extends AppCompatActivity
        implements BottomNavigationView.OnNavigationItemSelectedListener {
    private static MainActivity instance;
    private TextView txtTitle;
    public ImageView btnSearch;
    public ImageView btnNotification;
    public BottomNavigationView bottomNav;
    public SwipeRefreshLayout refreshLayout;
    public RelativeLayout toolBar;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        instance = this;
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        ButterKnife.bind(this);
        txtTitle = findViewById(R.id.txt_title);
        btnSearch = findViewById(R.id.btn_search);
        btnNotification = findViewById(R.id.btn_notify);
        bottomNav = findViewById(R.id.bottom_nav);
        refreshLayout = findViewById(R.id.refresh_layout);
        toolBar = findViewById(R.id.tb);
        refreshLayout.setRefreshing(false);
        bottomNav.setOnNavigationItemSelectedListener(this);
        bottomNav.setSelectedItemId(R.id.mnu_home);
        visibleBadge(R.id.mnu_bag);
    }

    @OnClick(R.id.btn_back)
    void onBackClicked(){
        super.onBackPressed();
    }

    public static synchronized MainActivity getInstance() {
        return instance;
    }

    public void setTitleActionBar(String titleActionBar){
        txtTitle.setText(titleActionBar);
    }

    public void loadFragment(Fragment fragment){
        FragmentUtil.replaceFragmentNonBackStack(this, fragment,false);
    }

    @Override
    public boolean onNavigationItemSelected(@NonNull MenuItem item) {
        switch (item.getItemId()){
            case R.id.mnu_home:
                loadFragment(new HomeFragment());
                return true;

            case R.id.mnu_bag:
                loadFragment(new OrderFragment());
                return true;

            case R.id.mnu_account:
                loadFragment(new PaymentFragment());
                return true;
        }
        return false;
    }

    public void visibleBadge(int menuItemId){
        BadgeDrawable badgeDrawable = bottomNav.getOrCreateBadge(menuItemId);
        badgeDrawable.setVisible(true);
        badgeDrawable.setBackgroundColor(getResources().getColor(android.R.color.holo_red_light));
        badgeDrawable.setNumber(8);
    }

    public void goneBadge(int menuItemId){
        BadgeDrawable badgeDrawable = bottomNav.getBadge(menuItemId);
        if (badgeDrawable != null) {
            badgeDrawable.setVisible(false);
        }
    }
}

File: app/src/main/java/com/lamnt/foodorder/view/staff/activity/MainActivity.kt
package com.lamnt.foodorder.view.staff.activity

import android.os.Bundle
import android.view.MenuItem
import androidx.fragment.app.Fragment
import com.google.android.material.bottomnavigation.BottomNavigationView
import com.lamnt.foodorder.R
import com.lamnt.foodorder.databinding.ActivityMainBinding
import com.lamnt.foodorder.utils.ActivityUtil.replaceFragmentNonBackStack
import com.lamnt.foodorder.view.base.BaseActivity
import com.lamnt.foodorder.view.staff.fragment.foody.HomeFragment
import com.lamnt.foodorder.view.staff.fragment.foody.OrderFragment
import com.lamnt.foodorder.view.staff.fragment.payment.PaymentFragment

class MainActivity : BaseActivity<ActivityMainBinding>(), BottomNavigationView.OnNavigationItemSelectedListener {
    override fun onCreate(savedInstanceState: Bundle?) {
        instance = this
        super.onCreate(savedInstanceState)
        with(viewBinding.bottomNav) {
            this.setOnNavigationItemSelectedListener(this@MainActivity)
            this.selectedItemId = R.id.mnu_home
        }
        visibleBadge(R.id.mnu_bag)
    }
    fun setTitleActionBar(titleActionBar: String?) {
        viewBinding.txtTitle.text = titleActionBar
    }

    private fun loadFragment(fragment: Fragment?) {
        replaceFragmentNonBackStack(this, fragment!!, false)
    }

    override fun onNavigationItemSelected(item: MenuItem): Boolean {
        when (item.itemId) {
            R.id.mnu_home -> {
                loadFragment(HomeFragment())
                return true
            }
            R.id.mnu_bag -> {
                loadFragment(OrderFragment())
                return true
            }
            R.id.mnu_account -> {
                loadFragment(PaymentFragment())
                return true
            }
        }
        return false
    }

    private fun visibleBadge(menuItemId : Int) {
        val badgeDrawable = viewBinding.bottomNav.getOrCreateBadge(menuItemId)
        badgeDrawable.isVisible = true
        badgeDrawable.backgroundColor = resources.getColor(android.R.color.holo_red_light)
        badgeDrawable.number = 8
    }

    fun goneBadge(menuItemId: Int) {
        val badgeDrawable = viewBinding.bottomNav.getBadge(menuItemId)
        if (badgeDrawable != null) {
            badgeDrawable.isVisible = false
        }
    }

    companion object {
        @JvmStatic
        @get:Synchronized
        var instance: MainActivity? = null
            private set
    }

    override fun setLayoutId(): Int = R.layout.activity_main

    override fun initViewModel() {

    }

    override fun initView() {

    }

    override fun initDataBinding() {

    }

    override fun initData() {

    }

    override fun initObserver() {

    }
}